LogosLinuxInstaller.py

```diff
@@ -183,57 +189,60 @@ def parse_args(args, parser):
-                raise argparse.ArgumentTypeError(f"Invalid file path: '{config.APPIMAGE_FILE_PATH}'. File does not exist.")
-            if not utils.check_appimage(config.APPIMAGE_FILE_PATH):
-                raise argparse.ArgumentTypeError(f"{config.APPIMAGE_FILE_PATH} is not an AppImage.")
-            config.ACTION = utils.set_appimage_symlink
-        else:
-            msg.logos_error("The command you requested is disabled. The configured install was not created using an AppImage.")
-    elif args.get_winetricks:
-        config.ACTION = control.get_winetricks
-    elif args.run_winetricks:
-        config.ACTION = wine.run_winetricks
-    elif args.toggle_app_logging:
-        config.ACTION = wine.switch_logging
-    elif args.create_shortcuts:
-        config.ACTION = installer.create_shortcuts
-    elif args.remove_install_dir:
-        config.ACTION = control.remove_install_dir
-    else: # default function if app is installed
+    actions = {
+        'install_app': installer.install,
+        'run_installed_app': wine.run_logos,
+        'run_indexing': wine.run_indexing,
+        'remove_library_catalog': control.remove_library_catalog,
+        'remove_index_files': control.remove_all_index_files,
+        'edit_config': control.edit_config,
+        'install_dependencies': utils.check_dependencies,
+        'backup': control.backup,
+        'restore': control.restore,
+        'update_latest_appimage': utils.update_to_latest_recommended_appimage,
+        'set_appimage': utils.set_appimage_symlink,
+        'get_winetricks': control.get_winetricks,
+        'run_winetricks': wine.run_winetricks,
+        'toggle_app_logging': wine.switch_logging,
+        'create_shortcuts': installer.create_shortcuts,
+        'remove_install_dir': control.remove_install_dir,
+    }
+
+    b = False
+    for arg, action in actions.items():
+        if getattr(args, arg):
+            if arg == "update_latest_appimage" or arg == "set_appimage":
+                logging.debug("Running an AppImage command.")
+                if config.WINEBIN_CODE != "AppImage" and config.WINEBIN_CODE != "Recommended":
+                    config.ACTION = "disabled"
+                    logging.debug("AppImage commands not added since WINEBIN_CODE != (AppImage|Recommended)")
+                    b = True
+                    break
+            if arg == "set_appimage":
+                config.APPIMAGE_FILE_PATH = getattr(args, arg)[0]
+                if not utils.file_exists(config.APPIMAGE_FILE_PATH):
+                    raise argparse.ArgumentTypeError(
+                        f"Invalid file path: '{config.APPIMAGE_FILE_PATH}'. File does not exist.")
+                if not utils.check_appimage(config.APPIMAGE_FILE_PATH):
+                    raise argparse.ArgumentTypeError(f"{config.APPIMAGE_FILE_PATH} is not an AppImage.")
+            config.ACTION = action
+            b = True
+            break
+    if not b:
         config.ACTION = run_control_panel
```

Comment:
I see where you're going with this, and I think it looks pretty good. My one suggestion is that I think we can avoid having to use `b` if we just start with `config.ACTION = None`. Then the check at the end can just be:
```
if config.ACTION is None:
    config.ACTION = run_control_panel
```

Reply:
Ah, that's good.